src/imitation/policies/trainer.py

```diff
@@ -0,0 +1,83 @@
+"""Training policies with a specifiable reward function and collect trajectories."""
+from typing import List, Union
+
+from stable_baselines3.common.base_class import BaseAlgorithm
+from stable_baselines3.common.vec_env import VecEnv
+
+from imitation.data import types
+from imitation.data.wrappers import BufferingWrapper
+from imitation.rewards.common import RewardFn
+from imitation.rewards.reward_nets import RewardNet
+from imitation.util.reward_wrapper import RewardVecEnvWrapper
+
+
+class AgentTrainer:
+    """Wrapper for training an SB3 algorithm on an arbitrary reward function.
+
+    TODO(ejnnr): For preference comparisons, we might want to allow something more
```

Comment:
You can make them separate functions without being separate computation. The `train()` can just cache samples that can be returned by `sample()` (maybe with a fallback if there's no samples in the buffer when `sample()` is called?)